uint16 message_id
uint8 event_type
uint8 event_id
uint8 all_events
---
Event[] events